opm/grid/MinpvProcessor.cpp

```diff
@@ -287,20 +288,14 @@ MinpvProcessor::process(const std::vector<double>& thickness,
                         // and sum of gaps is below threshold
                         const std::array<double, 8> cz_below = getCellZcorn(ii, jj, kk_iter, zcorn);
                         const std::array<double, 8> cz_above = getCellZcorn(ii, jj, k_above, zcorn);
-                        nnc_allowed = nnc_allowed && (computeGap(cz_above, cz_below) < max_gap);
+                        // top cell might not have been inspected for option 4 ALL before
+                        option4ALLZero = option4ALLZero || (!permz.empty() && permz[c_above] == 0.0) || multz(c_above) == 0.0;
+                        nnc_allowed = nnc_allowed && (computeGap(cz_above, cz_below) < max_gap) && (!pinchOption4ALL || !option4ALLZero) ;
```

Comment:
Yes that is intended, because PERMZ and MULTZ only play a role for item 4 all. Only in this case the transmissibiities of the cells between the active ones are used. If one of those is zero (due to PERMZ or MULTZ), we omit the NNC.